### ISSUE TITLE
RPC Filter parameters ignored; events from current block always returned

### ISSUE THREAD

**Issue by DefiDebauchery** [crypto-org-chain/cronos#226]
**Describe the bug**
When using event filters (`eth_newFilter`), neither block nor topic arguments are honored; all(?) events from the current block are returned.

**To Reproduce**
See code snippet below. 

**Expected behavior**
Event logs are returned relevant to the block(s) and topic(s) requested.

**Actual behavior**
All event logs from the current block are returned, including Transfers and other unrequested events.

**Additional context**
python 3.9 with web3.py installed

Added similar BSC network call for reference/control

```py
from web3 import Web3

cronos_block = 204485  # Random block in the past
topic = '0x4d6ce1e535dbade1c23defba91e23b8f791ce5edc0cc320257a2b364e4e38426'
cronos = Web3(Web3.HTTPProvider('https://evm-cronos.crypto.org'))

test_filter = cronos.eth.filter({'fromBlock': cronos_block, 'toBlock': cronos_block, 'topics': [topic]})

print('Cronos Target Block', target_block, '| Received Block', test_filter.get_all_entries()[0].blockNumber)

bsc_block = 12935778
bsc = Web3(Web3.HTTPProvider('https://bsc-dataseed.binance.org'))
test_filter = bsc.eth.filter({'fromBlock': bsc_block, 'toBlock': bsc_block, 'topics': [topic]})

print('BSC Target Block', target_block, '| Received Block', test_filter.get_all_entries()[0].blockNumber)
```

outputs
```
Cronos Target Block 204485 | Received Block 264670
BSC Target Block 12935778 | Received Block 12935778
```

**Comment by tomtau** [comment]
it looks like an issue on Ethermint, re-opened here https://github.com/tharsis/ethermint/issues/792

**Comment by DefiDebauchery** [comment]
Hi @tomtau - Looks like this has been fixed upstream (https://github.com/tharsis/ethermint/pull/865). Hoping this can be live on Cronos soon, it will open up a lot of important analytics work for projects!

**Comment by yihuang** [comment]
> Hi @tomtau - Looks like this has been fixed upstream ([tharsis/ethermint#865](https://github.com/tharsis/ethermint/pull/865)). Hoping this can be live on Cronos soon, it will open up a lot of important analytics work for projects!

I think the fix has already been included in Cronos releases, closing for now, feel free to reopen if still exists.
